### DISCUSSION TITLE
trying to use an Oak-D-Lite AI camera via USB...

### DISCUSSION THREAD

**Discussion by FolsomMike** [dorssel/usbipd-win#392] [General]
I have worked through all issues up to the point of getting the demo program to see and attempt to connect to the camera.

Then I get the following error:

     RuntimeError: Failed to find device after booting, error message: X_LINK_DEVICE_NOT_FOUND

There is very little information on the internet regarding the problem, and I am not sure where to begin.

Does this sound like a Windows, WSL, usbipd-win, Oak, or Linux issue?

I have seen a few comments that WSL works in specific use-cases (programming Arduinos, etc). Does something special have to be done to get it to work with each different device?

**Comment by dorssel** [comment]
If you're seeing the device with `lsusb` and *not* seeing errors in syslog, then it is most likely a missing driver. See https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-usbip-enabled-wsl-2-kernel on how to add drivers to the WSL kernel. Webcams most definitely need drivers that are not present in the default WSL kernel. If you haven't compiled your own kernel yet, then that is probably it.

**Reply by FolsomMike** [reply]
Turns out that the camera reconnects as a different device and name after the host code has rebooted it...which happens every time the program is run. I solved the problem by writing a script in WSL that continuously attaches any new USB device with a specified name to the open WSL.

Works great!

I have seen other devices which reconnect with different names, such as some Arduino boards. This trick would probably work for those as well.

I will post the script soon.

**Reply by vmchk** [reply]
Hey there!

I've been encountering the same problem myself. In my case, I wrote a script that attempts to connect all devices from the oak vendor. While I’m able to connect, I'm facing an issue where the depthai library throws an error indicating "X_LINK_DEVICE_ALREADY_IN_USE."

Could you please share the script that you used to continuously attach any new USB device with a specified name to the open WSL? It would be greatly appreciated!

Looking forward to your response!

**Reply by FolsomMike** [reply]
https://gitlab.com/spud-the-bot/wsl-usb-attach-daemon-for-oak-d-lite-cameras

See the readme.md file for instructions.
